Sources/CassandraClient/Data.swift

```diff
@@ -283,6 +308,27 @@ extension CassandraClient {
             cass_value_is_null(self.rawPointer) == cass_true
         }
     }
+
+    /// A reusable page token that can be used by `Statement` to resume querying
+    /// at a specific position.
+    public struct PagingStateToken: ContiguousBytes, Sequence {
+        public typealias Element = UInt8
+        public typealias Iterator = Array<Element>.Iterator
+
+        public func makeIterator() -> Iterator {
+            self.token.makeIterator()
+        }
+
+        private let token: [UInt8]
+
+        public init(token: [UInt8]) {
+            self.token = token
+        }
+
+        public func withUnsafeBytes<R>(_ body: (UnsafeRawBufferPointer) throws -> R) rethrows -> R {
```

Comment:
how is this used by clients? ideally we reduce unsafe API surface area

Reply:
oh I think I see, this is used as an "opaque" entity the users pass down to `setPagingStateToken`, I guess what we can do to hide some of the details is to create a "guts" internal type and use that inside  `setPagingStateToken` eg `token.guts` etc

Reply:
I actually want to use this feature myself externally, but doing it safely like the Java implementation. So I need access to the guts.

Reply:
Unfortunately, building that safety into the library itself would have necessitated pulling in additional dependencies and increasing the footprint of the library considerably, so I wanted to keep it simple here and allow users to use the token in a way that makes sense for their application.

Reply:
can you shed more light on what that could look like? generally ew prefer not promote unsafe APIs and here the only public API is unsafe

Reply:
What the Java implementation does is take the paging token, along with the query, and all the bound parameters, and hashes them together. It then also encodes the protocol version in the token. That way the token can only be used on the exact same query it came from, along with the same protocol it came from.

I originally had implemented that, but I needed a hashing algorithm to use. There doesn't appear to be anything within the Swift standard library, so I brought in SwiftCrypto. Unfortunately that requires a higher platform target, so I would have to bump that as well.

That seemed like a lot to change just to support this paging token. Maybe there's a better option I'm not thinking of?

Otherwise we could just expose the raw byte array and get rid of the contiguous bytes implementation. 

What do you think?